processing/src/main/java/org/apache/druid/segment/ColumnCardinalityInspector.java

```diff
@@ -19,7 +19,14 @@
 
 package org.apache.druid.segment;
 
-public interface CursorHolderFactory
+/**
+ * {@link ColumnInspector} + value cardinality. This can probably be merged into {@link ColumnInspector} at some point
+ * in the future.
+ */
+public interface ColumnCardinalityInspector extends ColumnInspector
 {
-  CursorHolder makeCursorHolder(CursorBuildSpec spec);
+  default int getColumnValueCardinality(String column)
```

Comment:
Should have javadocs at least describing what scenarios can result in `DimensionDictionarySelector.CARDINALITY_UNKNOWN`.

Reply:
this is removed instead, though did javadoc the similar method in the `GroupingSelector` that somewhat replaced this interface